internal/server/statvar/fetcher/formula_fetcher.go

```diff
@@ -0,0 +1,149 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetcher
+
+import (
+	"context"
+	"net/http"
+	"sort"
+
+	pbv2 "github.com/datacommonsorg/mixer/internal/proto/v2"
+	"github.com/datacommonsorg/mixer/internal/server/resource"
+	v1pv "github.com/datacommonsorg/mixer/internal/server/v1/propertyvalues"
+	"github.com/datacommonsorg/mixer/internal/store"
+	"github.com/datacommonsorg/mixer/internal/util"
+	"golang.org/x/sync/errgroup"
+)
+
+// V2Node for Remote Mixer.
+func FetchRemoteNode(
+	ctx context.Context,
+	metadata *resource.Metadata,
+	req *pbv2.NodeRequest,
+) (*pbv2.NodeResponse, error) {
+	errGroup, _ := errgroup.WithContext(ctx)
+	remoteResponseChan := make(chan *pbv2.NodeResponse, 1)
+	errGroup.Go(func() error {
+		remoteResp := &pbv2.NodeResponse{}
+		err := util.FetchRemote(metadata, &http.Client{}, "/v2/node", req, remoteResp)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err := errGroup.Wait(); err != nil {
+		return nil, err
+	}
+	close(remoteResponseChan)
+	return <-remoteResponseChan, nil
+}
+
+// FetchFormulas fetches StatisticalCalculations from storage and returns a map of SV -> inputPropertyExpressions.
+func FetchFormulas(
+	ctx context.Context,
+	store *store.Store,
+	metadata *resource.Metadata,
+) (map[string][]string, error) {
+	result := map[string][]string{}
+	svToCalculations := map[string][]string{}
+	calculationToFormulas := map[string][]string{}
+	// Fetch for BT and SQL.
+	data, _, err := v1pv.Fetch(
+		ctx,
+		store,
+		[]string{"StatisticalCalculation"},
+		[]string{"typeOf"},
+		0,
+		"",
+		"in",
+	)
+	if err != nil {
+		return nil, err
+	}
+	statisticalCalculations := []string{}
+	for _, nodes := range data["StatisticalCalculation"]["typeOf"] {
+		for _, node := range nodes {
+			statisticalCalculations = append(statisticalCalculations, node.Dcid)
+		}
+	}
+	data, _, err = v1pv.Fetch(
+		ctx,
+		store,
+		statisticalCalculations,
+		[]string{"outputProperty", "inputPropertyExpression"},
+		0,
+		"",
+		"out",
+	)
+	if err != nil {
+		return nil, err
+	}
+	for dcid, properties := range data {
+		for _, nodes := range properties["outputProperty"] {
+			for _, node := range nodes {
+				svToCalculations[node.Dcid] = append(svToCalculations[node.Dcid], dcid)
+			}
+		}
+		for _, nodes := range properties["inputPropertyExpression"] {
+			for _, node := range nodes {
+				calculationToFormulas[dcid] = append(calculationToFormulas[dcid], node.Value)
+			}
+		}
+	}
+	// Fetch for Remote Mixer.
+	if metadata.RemoteMixerDomain != "" {
+		req := &pbv2.NodeRequest{
+			Nodes:    []string{"StatisticalCalculation"},
+			Property: "<-typeOf",
+		}
+		statisticalCalculationResp, err := FetchRemoteNode(ctx, metadata, req)
+		if err != nil {
+			return nil, err
+		}
+		statisticalCalculations := []string{}
+		for _, node := range statisticalCalculationResp.Data["StatisticalCalculation"].Arcs["typeOf"].Nodes {
```

Comment:
should have test case to check empty response and foobar input

Reply:
I wasn't entirely sure what you meant here, but I added a check for empty response